### DISCUSSION TITLE
Homework Submission

### DISCUSSION THREAD

**Discussion by safdarmt** [cooperrc/computational-mechanics#210] [General]
I have two questions regarding homework submission in general: 

1. In what format do we need to download the Jupyter Notebook to be able to submit it in google classroom?
2. When a problem asks for the solution to be timed, do we just need to include the %%time command in our code? Or, do we need to time it and then also use a print statement to print the amount of time for said problem?

Please advise!

**Comment by cooperrc** [comment]
Thanks for kicking off this discussion!

> 1. In what format do we need to download the Jupyter Notebook to be able to submit it in google classroom?

Final format submission is pdf:
- click File -> Download as -> HTML
- open the .html file
- print to pdf

If you want to use $$\LaTeX$$, you can "download as pdf", but you usually get a `Error 500` response with Tex compiling

> 2. When a problem asks for the solution to be timed, do we just need to include the %%time command in our code? Or, do we need to time it and then also use a print statement to print the amount of time for said problem?

You can time cells with 

```python
%% time
x = 1
for i in range(10):
  x *= i
```

*Here I'm timing a factorial operation*

It automatically prints the result of the timed operation. It changes each time you run it, so if you want a more stable result, use `%%timeit` to get an average over ~100,000 runs of the cell

e.g. 
```python
%% timeit
x = 1
for i in range(10):
  x *= i
```
